java/google/registry/rde/imports/XjcToDomainResourceConverter.java

```diff
@@ -293,16 +238,34 @@ private static HistoryEntry createHistoryEntry(XjcRdeDomain domain) {
   }
 
   /** Converts {@link XjcRdeDomainTransferDataType} to {@link TransferData}. */
-  private static TransferData convertDomainTransferData(XjcRdeDomainTransferDataType data) {
+  private static TransferData convertDomainTransferData(XjcRdeDomainTransferDataType data,
+      DateTime domainExpiration) {
     if (data == null) {
       return TransferData.EMPTY;
     }
+    // If the transfer is pending, calculate the number of years to add to the domain expiration
+    // on approval of the transfer.
+    TransferStatus transferStatus = TRANSFER_STATUS_MAPPER.xmlToEnum(data.getTrStatus().value());
+    // Get new expiration date
+    DateTime newExpirationTime = domainExpiration;
+    if (transferStatus == TransferStatus.PENDING) {
+      // Default to domain expiration time plus one year if no expiration is specified
```

Comment:
Good question. I noticed that the schema shows exDate as a nullable field, and I have encountered real deposit files with this attribute missing. With a real lack of documentation describing what a missing exDate means, it seems reasonable that the default should be one year past the domain expiration. However this has the same problem as you mentioned above, where there is no cap set at 10 years.